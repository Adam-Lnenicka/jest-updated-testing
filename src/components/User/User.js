import React, { forwardRef, useEffect, useState } from "react";

import fetchAuthors from "./_fetchUsers";

const User = forwardRef(({ disabled }, ref) => {
  const [authors, setAuthors] = useState(null);

  useEffect(() => {
    fetchAuthors().then((authors) => {
      setAuthors(authors);
    });
  }, []);

  return (
    <div disabled={disabled || !authors} name="plan" ref={ref}>
      {!authors ? (
        <div value="">Loading...</div>
      ) : (
        <div className="flex">
          {authors.map(({ id, name, email, username }) => (
            <div key={id} value={id} className="card">
              <h2>User: {name}</h2>
              <p>Email: {email}</p>
              <p>Username: {username}</p>
            </div>
          ))}
        </div>
      )}
    </div>
  );
});

export default User;
