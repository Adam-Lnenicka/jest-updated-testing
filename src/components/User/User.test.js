import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import React from "react";
import { act } from "react-dom/test-utils";
import User from "./User";

jest.mock("./_fetchUsers", () => {
  return {
    __esModule: true,
    default: async () => [
      {
        id: 1,
        name: "Test Author 1",
        email: "Test email",
        username: "Test username",
      },
      {
        id: 2,
        name: "Test Author 2",
        email: "Test email",
        username: "Test username",
      },
    ],
  };
});

test("matches snapshot", async () => {
  const Users = User;

  let wrapper;

  await act(async () => {
    wrapper = mount(<Users />);
  });

  wrapper.update();

  expect(toJSON(wrapper)).toMatchSnapshot();
});
