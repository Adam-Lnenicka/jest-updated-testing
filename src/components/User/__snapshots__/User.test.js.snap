// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<ForwardRef>
  <div
    disabled={false}
    name="plan"
  >
    <div
      className="flex"
    >
      <div
        className="card"
        key="1"
        value={1}
      >
        <h2>
          User: 
          Test Author 1
        </h2>
        <p>
          Email: 
          Test email
        </p>
        <p>
          Username: 
          Test username
        </p>
      </div>
      <div
        className="card"
        key="2"
        value={2}
      >
        <h2>
          User: 
          Test Author 2
        </h2>
        <p>
          Email: 
          Test email
        </p>
        <p>
          Username: 
          Test username
        </p>
      </div>
    </div>
  </div>
</ForwardRef>
`;
